tree.go

```diff
@@ -930,13 +1037,54 @@ func (n *LeafNode) Hash() *Fr {
 
 func (n *LeafNode) Commitment() *Point {
 	if n.commitment == nil {
-		panic("nil commitment")
+		n.Commit()
 	}
 	return n.commitment
 }
 
-func (n *LeafNode) Commit() *Point {
-	return n.commitment
+var frPool = sync.Pool{
+	New: func() any {
+		ret := make([]Fr, NodeWidth)
+		return &ret
+	},
+}
+
+func (leaf *LeafNode) Commit() *Point {
+	if leaf.commitment == nil {
+		// Initialize the commitment with the extension tree
+		// marker and the stem.
+		count := 0
+		polyp, c1polyp := frPool.Get().(*[]Fr), frPool.Get().(*[]Fr)
```

Comment:
When `Commit()` a leaf, we also leverage the `[]Fr` pool that we use in other places. These polynomials are ephemeral for calculations, so we avoid generating garbage.